src/System.Memory/tests/ReadOnlySpan/BinarySearch.cs

```diff
@@ -0,0 +1,190 @@
+                (new double[] { 1.0, 2.0, 4.0, 5u }, 5.0, 3),
+                (new double[] { 1.0, 2.0, 4.0, 5u }, 6.0, -5),
+            };
+        public static readonly TheoryData<(string[] Array, string Value, int ExpectedIndex)> StringCases =
+            new TheoryData<(string[] Array, string Value, int ExpectedIndex)> {
+                (new string[] { }, "a", -1),
+                (new string[] { "b" }, "a", -1),
+                (new string[] { "b" }, "b", 0),
+                (new string[] { "b" }, "c", -2),
+                (new string[] { "b", "c", "e", "f" }, "a", -1),
+                (new string[] { "b", "c", "e", "f" }, "b", 0),
+                (new string[] { "b", "c", "e", "f" }, "c", 1),
+                (new string[] { "b", "c", "e", "f" }, "d", -3),
+                (new string[] { "b", "c", "e", "f" }, "e", 2),
+                (new string[] { "b", "c", "e", "f" }, "f", 3),
+                (new string[] { "b", "c", "e", "f" }, "g", -5),
+            };
+
+        [Theory, MemberData(nameof(UIntCases))]
+        public static void BinarySearch_UInt(
+            (uint[] Array, uint Value, int ExpectedIndex) c)
+        {
+            TestOverloads(c.Array, c.Value, c.ExpectedIndex);
+        }
+
+        [Theory, MemberData(nameof(DoubleCases))]
+        public static void BinarySearch_Double(
+            (double[] Array, double Value, int ExpectedIndex) c)
+        {
+            TestOverloads(c.Array, c.Value, c.ExpectedIndex);
+        }
+
+        [Theory, MemberData(nameof(StringCases))]
+        public static void BinarySearch_String(
+            (string[] Array, string Value, int ExpectedIndex) c)
+        {
+            TestOverloads(c.Array, c.Value, c.ExpectedIndex);
+        }
+
+        [Fact]
+        public static void BinarySearch_NullComparableThrows()
+        {
+            Assert.Throws<ArgumentNullException>(() => new Span<int>(new int[] { }).BinarySearch<int>(null));
+            Assert.Throws<ArgumentNullException>(() => new ReadOnlySpan<int>(new int[] { }).BinarySearch<int>(null));
+            Assert.Throws<ArgumentNullException>(() => new Span<int>(new int[] { }).BinarySearch<int, IComparable<int>>(null));
+            Assert.Throws<ArgumentNullException>(() => new ReadOnlySpan<int>(new int[] { }).BinarySearch<int, IComparable<int>>(null));
+        }
+
+        // TODO: Revise whether this should actually throw
+        [Fact]
+        public static void BinarySearch_NullComparerThrows()
+        {
+            Assert.Throws<ArgumentNullException>(() => new Span<int>(new int[] { }).BinarySearch<int, IComparer<int>>(0, null));
+            Assert.Throws<ArgumentNullException>(() => new ReadOnlySpan<int>(new int[] { }).BinarySearch<int, IComparer<int>>(0, null));
+        }
+
+        [Fact]
+        // TODO: Does it need to be OuterLoop, it's pretty fast, compared to total time
+        //[OuterLoop]
+        // TODO: Do we need to be platform specific?
```

Comment:
Yes. We can't consistently run such tests on Linux.
See rationale: https://github.com/dotnet/corefx/blob/master/src/System.Memory/tests/Span/Clear.cs#L229